server/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -614,15 +616,63 @@ public void messageReceived(StartedShardEntry request, TransportChannel channel,
                 request,
                 ClusterStateTaskConfig.build(Priority.URGENT),
                 shardStartedClusterStateTaskExecutor,
-                e -> {
-                    if (e instanceof FailedToCommitClusterStateException || e instanceof NotMasterException) {
-                        logger.debug(new ParameterizedMessage("failure during [shard-started {}]", request), e);
-                    } else {
-                        logger.error(new ParameterizedMessage("unexpected failure during [shard-started {}]", request), e);
+                new ClusterStateTaskListener() {
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.error(
+                            () -> new ParameterizedMessage("{} unexpected failure while starting shard [{}]", request.shardId, request),
+                            e
+                        );
+                        try {
+                            channel.sendResponse(e);
+                        } catch (Exception channelException) {
+                            channelException.addSuppressed(e);
+                            logger.warn(
```

Comment:
This is pretty much what `ChannelActionListener` does, maybe we should just use that?

Reply:
Yea right lets do that :)